ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -278,13 +234,20 @@ getQuadratureKey(const QuadratureType quad_type,
                  const double dx_min)
 {
     const ElemType elem_type = elem->type();
+#ifndef NDEBUG
     TBOX_ASSERT(elem->p_level() == 0); // higher levels are not implemented
+#endif
     if (use_adaptive_quadrature)
     {
-        const double hmax = get_elem_hmax(elem, X_node);
-        const int min_pts = elem->default_order() == FIRST ? 2 : 3;
-        const int npts = std::max(min_pts,
-                                  int(std::ceil(point_density * hmax / dx_min)));
+        const double hmax = get_max_edge_length(elem, X_node);
+        int npts = int(std::ceil(point_density * hmax / dx_min));
```

Comment:
You don't like my use of the ternary operator?

Reply:
I don't mind :) I changed this bit to avoid one more virtual function call (to `Elem::default_order)`.